src/main/java/edu/oregonstate/cs361/battleship/ComputerAI.java

```diff
@@ -1,66 +1,235 @@
+        ArrayList<Point> adjacentSpots;
+
+        if(cpuHardDifficulty == true) {
+            //All hits that aren't associated with destroyed ships.
+            injuredShips = hitLiveShipChecker(cpuModel);
+
+            //If we've detected a live ship, hunt it down.
+            if(injuredShips.size() != 0) {
+                adjacentSpots = listRelevantViableCoords(injuredShips, cpuModel);
+
+                //An optimization could be inserted here to shoot in the path as the rest of the hits.
+                cpuTargetFired.setRow(adjacentSpots.get(0).getRow());
+                cpuTargetFired.setCol(adjacentSpots.get(0).getCol());
+            }
+
+            //Else search for a new target.
+            else {
+                do {
+                    //Can be refined to shoot in every other space.
+                    rawCoords = pos.nextInt(100);
+                    cpuFireRow = rawCoords / 10;
+                    cpuFireCol = rawCoords % 10;
+                }while(fireOverlapChecker(cpuModel, cpuFireRow, cpuFireCol) == true);
+
+                cpuTargetFired.setRow(cpuFireRow);
+                cpuTargetFired.setCol(cpuFireCol);
+            }
+        }
+
+        else {
+            do {
+                rawCoords = pos.nextInt(100);
+                cpuFireRow = rawCoords / 10;
+                cpuFireCol = rawCoords % 10;
+            }while(fireOverlapChecker(cpuModel, cpuFireRow, cpuFireCol) == true);
+
+            cpuTargetFired.setRow(cpuFireRow);
+            cpuTargetFired.setCol(cpuFireCol);
+        }
+
+        return cpuTargetFired;
+    }
+
+    private ArrayList<Point> listRelevantViableCoords(ArrayList<Point> injuredShips, BattleshipModel cpuModel) {
+        ArrayList<Point> adjCoords = new ArrayList<Point>();
+        Point temp;
 
-        /*
-        Currently this AI will loop from left-top to every other space.
-         */
+        for(int i = 0; i < injuredShips.size(); i++) {
+            temp = injuredShips.get(i);     //Reset Temp
+            temp.setCol(temp.getCol() - 1); //West
+            adjCoords.add(temp);
 
-        // If statement for starting state at Point (1,1)
-        if (cpuFireRow == 1 && cpuFireCol == 1) {
-            cpuFireCol += 2;
-            return cpuTargetFired;
+            temp.setCol(temp.getCol() + 2); //East
+            adjCoords.add(temp);
+
+            temp = injuredShips.get(i);     //Reset Temp
+            temp.setRow(temp.getRow() - 1); //North
+            adjCoords.add(temp);
+
+            temp.setRow(temp.getRow() + 2); //South
+            adjCoords.add(temp);
+        }
+
+        adjCoords = cleanAdjacencyList(adjCoords, cpuModel.getComputermisses(), injuredShips);
+
+        //An optimization could be inserted here to shoot in the path as the rest of the hits.
+
+        return adjCoords;
+    }
+
+    private ArrayList<Point> cleanAdjacencyList(ArrayList<Point> dirtyAdjCoords, ArrayList<Point> misses, ArrayList<Point> hits) {
```

Comment:
These functions look like they are written in a clean fashion and appear to follow the coding style guidelines that were covered in class.